cli/src/pcluster/templates/pcluster_dashboard_builder.py

```diff
@@ -0,0 +1,586 @@
+                        cloudwatch_metric = cloudwatch.Metric(
+                            namespace=namespace,
+                            metric_name=metric_condition_params["metric"],
+                            dimensions={dimension_vol_name: getattr(volume, attribute_name, "")},  # TODO:FIXME
+                        )
+                        metric_list.append(cloudwatch_metric)
+                graph_widget = self._generate_graph_widget(metric_condition_params["title"], metric_list)
+                widgets_list.append(graph_widget)
+        return widgets_list
+
+    def _add_storage_widgets(self, metrics, storages_list, namespace, dimension_name, attribute_name):
+        widgets_list = []
+        for metrics_param in metrics:
+            metric_list = []
+            for metric in metrics_param["metrics"]:
+                for storage in storages_list:
+                    cloudwatch_metric = cloudwatch.Metric(
+                        namespace=namespace,
+                        metric_name=metric,
+                        dimensions={dimension_name: getattr(storage, attribute_name, "")},  # TODO:FIXME
+                    )
+                    metric_list.append(cloudwatch_metric)
+            graph_widget = self._generate_graph_widget(metrics_param["title"], metric_list)
+            widgets_list.append(graph_widget)
+        return widgets_list
+
+    def _add_head_node_instance_metrics_graphs(self, cloudwatch_dashboard):
+        # Create a text widget for subtitle "Head Node Instance Metrics"
+        self._add_text_widget("\n## Head Node Instance Metrics\n", cloudwatch_dashboard)
+
+        # Create EC2 metrics graph for head node instance
+        ec2_metrics = [
+            {"metrics": ["CPUUtilization"], "title": "CPU Utilization"},
+            {"metrics": ["NetworkPacketsIn", "NetworkPacketsOut"], "title": "Network Packets In/Out"},
+            {"metrics": ["NetworkIn", "NetworkOut"], "title": "Network In and Out"},
+            {"metrics": ["DiskReadBytes", "DiskWriteBytes"], "title": "Disk Read/Write Bytes"},
+            {"metrics": ["DiskReadOps", "DiskWriteOps"], "title": "Disk Read/Write Ops"},
+        ]
+        widgets_list = []
+        for metrics_param in ec2_metrics:
+            metrics_list = self._generate_ec2_metrics_list(metrics_param["metrics"])
+            graph_widget = self._generate_graph_widget(metrics_param["title"], metrics_list)
+            widgets_list.append(graph_widget)
+        cloudwatch_dashboard.add_widgets(*widgets_list)
+        self._update_coord_after_section(self.coord, self.graph_height)
+
+    def _add_ebs_metrics_graphs(self, cloudwatch_dashboard, ebs_metrics, conditional_ebs_metrics):
+        # Create a text widget for subtitle "EBS Metrics"
+        self._add_text_widget("\n## EBS Metrics\n", cloudwatch_dashboard)
+
+        # Create unconditional EBS metrics graph widgets
```

Comment:
minor: This comment can be moved right before:
`        widgets_list += self._add_storage_widgets(ebs_metrics, ebs_volumes_list, "AWS/EBS", "VolumeId", "volume_id")`